Lib/calendar.py

```diff
@@ -349,11 +349,27 @@ def formatday(self, day, weekday, width):
             s = '%2i' % day             # right-align single-digit days
         return s.center(width)
 
-    def formatweek(self, theweek, width):
+    def formatweek(self, theweek, width, *, highlight_day=None):
         """
         Returns a single week in a string (no newline).
         """
-        return ' '.join(self.formatday(d, wd, width) for (d, wd) in theweek)
+        if highlight_day:
+            from _colorize import get_colors
+
+            ansi = get_colors()
+            highlight = f"{ansi.BLACK}{ansi.BACKGROUND_YELLOW}"
+            reset = ansi.RESET
+        else:
+            highlight = reset = ""
+
+        return ' '.join(
+            (
+                f"{highlight}{self.formatday(d, wd, width)}{reset}"
+                if d == highlight_day
+                else self.formatday(d, wd, width)
+            )
+            for (d, wd) in theweek
+        )
```

Comment:
I also considered something like this, but no strong preference:
```suggestion
        if highlight_day:
            from _colorize import get_colors

            ansi = get_colors()
            highlight = f"{ansi.BLACK}{ansi.BACKGROUND_YELLOW}"
            reset = ansi.RESET

            return ' '.join(
                (
                    f"{highlight}{self.formatday(d, wd, width)}{reset}"
                    if d == highlight_day
                    else self.formatday(d, wd, width)
                )
                for (d, wd) in theweek
            )
        else:
            return ' '.join(
                self.formatday(d, wd, width) for (d, wd) in theweek
            )
```

Could also factor out a `return ' '.join(formatted_week)` to outside the `if`/`else`.